carapace-server/src/main/java/org/carapaceproxy/server/certificates/DynamicCertificatesManager.java

```diff
@@ -281,55 +280,28 @@ private void certificatesLifecycle() {
                         break;
                     }
                     case DNS_CHALLENGE_WAIT: { // waiting for full dns propagation
-                        LOG.log(Level.INFO, "DNS_CHALLENGE_WAIT for domain {0}.", domain);
+                        LOG.log(Level.INFO, "DNS CHALLENGE WAITING for domain {0}.", domain);
                         Dns01Challenge pendingChallenge = (Dns01Challenge) getChallengeFromCertificate(cert);
-                        if (pendingChallenge == null) {
```

Comment:
Temporary network error would wipe-out what done 'till this point. Better to avoid that.